### PR TITLE
update createLogger to sync, update express routes config, adjust https deps & types, fix tests

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
- update createLogger to sync, update express routes config, adjust https deps & types, fix tests

## Changelog
- update createLogger to sync, update express routes config, adjust https deps & types, fix tests

## Test Plan
- Updated Tests

### CODE REVIEW

File: packages/gasket-plugin-express/lib/create-servers.js [hunk 1]
```diff
@@ -18,6 +14,8 @@ module.exports = async function createServers(gasket, serverOpts) {
 
   const { config, logger } = gasket;
   const {
+    // May be needed later on
+    // eslint-disable-next-line no-unused-vars
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-express/lib/create-servers.js:18]: probably not needed, I'd go ahead and remove

File: packages/gasket-plugin-express/lib/create-servers.js [hunk 2]
```diff
@@ -131,9 +129,16 @@ module.exports = async function createServers(gasket, serverOpts) {
...
+    for (const route of routes) {
+      if (typeof route !== 'function') {
+        throw new Error('Route must be a function');
+      }
+
+      if (route.constructor.name === 'AsyncFunction') {
+        await route(app);
+      } else {
+        route(app);
+      }
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-express/lib/create-servers.js:141]: Should be ok just to await all these for simplicity
```suggestion
      await route(app);
```

**Review by kbader-godaddy — APPROVED** [review]

File: packages/gasket-plugin-logger/lib/index.js [hunk 3]
```diff
@@ -26,6 +26,8 @@ module.exports = {
     init(gasket) {
       // eslint-disable-next-line no-sync
       const loggers = gasket.execSync('createLogger');
+
+      if (loggers && loggers.some((logger) => logger && logger instanceof Promise)) throw new Error('createLogger hooks must be synchronous');
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-logger/lib/index.js:30]: nit for readability
```suggestion
      if (loggers && loggers.some((logger) => logger && logger instanceof Promise)) {
        throw new Error('createLogger hooks must be synchronous');
      }
```

File: packages/gasket-plugin-winston/lib/index.js [hunk 4]
```diff
@@ -24,7 +24,8 @@ module.exports = {
         configTransports = [new transports.Console()];
       }
 
-      const pluginTransports = await gasket.exec('winstonTransports');
+      // eslint-disable-next-line no-sync
+      const pluginTransports = gasket.execSync('winstonTransports');
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-winston/lib/index.js:28]: Adjust the metadata description also below
```diff
- method: 'exec',
+ method: 'execSync',
```

**Review by agerard-godaddy — APPROVED** [review]

**Pull request merged**